app/controllers/api/v1/groups_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Controller for managing groups in the API version 1 namespace.
+class Api::V1::GroupsController < ApplicationController
+  before_action :set_group, only: %i[show update destroy]
+
+  def index
+    @groups = Group.all
+    render json: @groups
```

Comment:
Ensure sensitive/unnecessary attributes are not exposed in any of the endpoints. You can try to make use of Active Model Serializers to expose only the required attributes